internal/codespaces/api/api_test.go

```diff
@@ -115,6 +115,145 @@ func TestListCodespaces_unlimited(t *testing.T) {
 	}
 }
 
+func createFakeSearchReposServer(t *testing.T, searchText string, responseRepos []*Repository) *httptest.Server {
+	return httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.URL.Path != "/search/repositories" {
+			t.Fatal("Incorrect path")
+		}
+
+		if r.URL.Query().Get("q") == "" {
```

Comment:
You could simplify this logic and print better error messages (with greater context about the request) thus:
```
query := r.URL.Query()
got := fmt.Sprintf("q=%q sort=%s per_page=%s", query.Get("q"), query.Get("sort"), query.Get("per_page"))
want := fmt.Sprintf("q=%q sort=stars per_page=7", searchText + " in:name")
if got != want {
    t.Errorf("for query, got %s, want %s", got, want)
}
```